core/templates/dev/head/domain/exploration/StatesObjectFactory.js

```diff
@@ -18,18 +18,112 @@
  */
 
 oppia.factory('StatesObjectFactory', [
-  'StateObjectFactory', function(StateObjectFactory) {
-    var createFromBackendDict = function(statesBackendDict) {
+  'StateObjectFactory', 'newStateTemplateService', 'INTERACTION_SPECS',
+  function(StateObjectFactory, newStateTemplateService, INTERACTION_SPECS) {
+    var States = function(states) {
+      var states = states;
```

Comment:
So I ended up just making it this because I couldn't figure out how to make states private without making StatesObjectFactory behave as a singleton as opposed to a factory.